### PR TITLE
Add new hook for reset actions

### PR DESCRIPTION
**Pull request by anxuae**
New hook to reset the configuration.

The place where the hook is implemented (``view_plugin.py``) is not optimum, if you have an other idea?

### CODE REVIEW

File: pibooth/booth.py [hunk 1]
```diff
@@ -423,7 +427,8 @@ def main():
         LOGGER.info("Listing all fonts available...")
         print_columns_words(get_available_fonts(), 3)
     elif options.reset:
-        config.save()
+        config.save(True)
```
Comment (werdeil) [inline pibooth/booth.py:430]: Same here to be easier to read I would put
> config.save(default=True)
even if I agree it is the same

File: pibooth/booth.py [hunk 2]
```diff
@@ -410,8 +410,12 @@ def main():
 
     # Update configuration with plugins ones
     plugin_manager.hook.pibooth_configure(cfg=config)
-    if not osp.isfile(config.filename):
-        config.save()
+
+    # Ensure config files are present in case of first pibooth launch
+    if not options.reset:
+        if not osp.isfile(config.filename):
+            config.save(True)
```
[the remark this answers is not in the document]
Comment (anxuae) [reply pibooth/booth.py]: done

File: pibooth/booth.py (hunk 1 above)
Comment (anxuae) [reply pibooth/booth.py]: done

File: pibooth/plugins/hookspecs.py [hunk 3]
```diff
@@ -13,15 +13,27 @@ def pibooth_configure(cfg):
...
     class.
 
-    :param cfg: loaded configuration
+    :param cfg: application configuration
+    """
+
+
+@hookspec
+def pibooth_reset(cfg, hard):
+    """Restore the default configuration.
```
[the remark this answers is not in the document]
Comment (anxuae) [reply pibooth/plugins/hookspecs.py]: Agree, I added details on the hook description

**Pull request merged**